### PR TITLE
logically seperate p2p config

### PR DESCRIPTION
**Pull request by cong-or**
Currently the p2p config has become cluttered with multiple behaviours; i.e internode relations, bootstrap info and other things. PR adds extra sub settings to logically separate the functionality for clarity and context.

### CODE REVIEW

File: src/jormungandr/jormungandr/src/settings/start/mod.rs [hunk 1]
```diff
@@ -288,21 +289,21 @@ fn generate_network(
 
     // TODO: do we want to check that we end up with a valid address?
```
Comment (kukkok3) [inline src/jormungandr/jormungandr/src/settings/start/mod.rs:290]: leftover comment

File: src/jormungandr/testing/hersir/src/builder/explorer.rs [hunk 2]
```diff
@@ -16,7 +16,13 @@ pub fn generate_explorer(
...
         explorer_listen_address: "127.0.0.1".to_string(),
-        node_address: settings.config.p2p.public_address.clone().to_http_addr(),
+        node_address: settings
+            .config
+            .p2p
+            .connection
+            .public_address
+            .clone()
+            .to_http_addr(),
         logs_dir: Some(Path::new("C:\\work\\iohk\\logs.txt").to_path_buf()),
```
Comment (kukkok3) [inline src/jormungandr/testing/hersir/src/builder/explorer.rs:26]: leftover path?

File: src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/public_traffic.rs [hunk 3]
```diff
@@ -94,8 +94,8 @@ fn public_gossip_rejection() {
     assert!(gossip_dropped);
 }
 
-#[ignore]
 #[test]
+#[ignore]
 pub fn test_public_node_cannot_publish() {
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/public_traffic.rs:99]: why this tests are in ignore?It is done on purpose?

File: src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/public_traffic.rs [hunk 4]
```diff
@@ -154,19 +154,19 @@ pub fn test_public_node_cannot_publish() {
...
-    gateway_node_config.p2p.whitelist = Some(whitelist.clone());
+    gateway_node_config.p2p.connection.whitelist = Some(whitelist.clone());
 
     let mut public_node_config = network_controller.node_config(PUBLIC_NODE).unwrap();
 
-    public_node_config.p2p.whitelist = Some(whitelist.clone());
+    public_node_config.p2p.connection.whitelist = Some(whitelist.clone());
 
     //
     //
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/public_traffic.rs:172]: lefovers //

File: src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/public_traffic.rs [hunk 5]
```diff
@@ -241,8 +241,8 @@ pub fn test_public_node_cannot_publish() {
     };
 }
 
-#[ignore]
 #[test]
+#[ignore]
```
Comment (kukkok3) [inline src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/public_traffic.rs:245]: same, why ignore?

File: src/jormungandr/testing/hersir/src/builder/explorer.rs (hunk 2 above)
Comment (cong-or) [reply src/jormungandr/testing/hersir/src/builder/explorer.rs:26]: This path was here before. Not sure why.

File: src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/public_traffic.rs (hunk 3 above)
Comment (cong-or) [reply src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/public_traffic.rs:99]: Yes, mainly because each test takes over 1 minute.

**Review by kukkok3 — APPROVED** [review]
LGTM

**Pull request merged**